### PR TITLE
Add doc comment

### PR DESCRIPTION
**Pull request by k-nasa**
## Description

I added a document.
I think that there is an inappropriate expression or wording, or an error in English, so please give me a review.
## Motivation and Context

close: #244

## How Has This Been Tested?

I generated a document and confirmed that there was no warning

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by k-nasa** [comment]
@prasannavl It seems that #255 has been approved. I will rebase after merge #255.

**Comment by fairingrey** [comment]
@k-nasa #255 was merged, feel free to rebase this whenever :+1:

**Comment by k-nasa** [comment]
@fairingrey I did rebase! Please give a review if you have time. 🙏

File: src/error.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+//! Module to export tide_core errors
+
+use core::pin::Pin;
+use futures::future::Future;
+
+pub use tide_core::error::{
+    EndpointResult, Error, ResponseExt, ResultDynErrExt, ResultExt, StringError,
+};
+
+pub(crate) type BoxTryFuture<T> = Pin<Box<dyn Future<Output = EndpointResult<T>> + Send + 'static>>;
```
Comment (yoshuawuyts) [inline src/error.rs]: This file doesn't seems unrelated to documentation; was this perhaps checked in accidentally?

Comment (k-nasa) [reply src/error.rs]: I'm sorry. I missed my self review. I fixed it!
